internal/backend/api.go

```diff
@@ -128,7 +142,73 @@ func (h *APIHandler) listAccessRequestHandler(ctx context.Context, input *ListAc
 
 // TODO implementation
 func (h *APIHandler) createAccessRequestHandler(ctx context.Context, input *CreateAccessRequestInput) (*CreateAccessRequestResponse, error) {
-	return nil, huma.Error501NotImplemented("not implemented")
+	// - TODO? Get current access request
+	//   - Return 400 if already exist
+	// - Get the access binding for role name
+	//   - return 404
+	// - Evaluate the if
+	// - Evaluate the template
+	// - Make sure the user group match
+	//   - Return 403
+	// - Create AR
+	//   - return 500 if any
+
+	appNamespace, appName, err := input.Application()
+	if err != nil {
+		return nil, huma.Error400BadRequest("error getting application name", err)
+	}
+
+	app, err := h.service.GetApplication(ctx, appName, appNamespace)
+	if err != nil {
+		return nil, huma.Error500InternalServerError("error getting application", err)
+	}
+	if app == nil {
+		return nil, huma.Error400BadRequest("invalid application", err)
+	}
+	projectName, ok, err := unstructured.NestedString(app.Object, "spec", "project")
+	if !ok {
+		return nil, huma.Error400BadRequest("invalid application spec", err)
+	}
+	if err != nil {
+		return nil, huma.Error400BadRequest("invalid application spec", err)
+	}
```

Comment:
This code is probably not required as the project name should already be provided in the input coming from the forwarded Argo CD headers.